zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -32,40 +44,87 @@
 import org.jboss.seam.annotations.Startup;
 import org.jboss.seam.annotations.intercept.BypassInterceptors;
 import org.jboss.seam.contexts.Contexts;
-import org.jboss.seam.core.Events;
-import org.jboss.seam.security.AuthorizationException;
-import org.jboss.seam.security.Configuration;
-import org.jboss.seam.security.Identity;
-import org.jboss.seam.security.NotLoggedInException;
-import org.jboss.seam.security.management.JpaIdentityStore;
+import org.zanata.exception.AuthorizationException;
+import org.zanata.exception.NotLoggedInException;
+import org.jboss.seam.web.Session;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import com.google.common.collect.Lists;
+import org.zanata.events.AlreadyLoggedInEvent;
+import org.zanata.events.LoginFailedEvent;
+import org.zanata.events.LoginSuccessfulEvent;
 import org.zanata.events.Logout;
+import org.zanata.events.NotLoggedInEvent;
 import org.zanata.model.HAccount;
 import org.zanata.model.HasUserFriendlyToString;
+import org.zanata.seam.security.ZanataJpaIdentityStore;
+import org.zanata.security.jaas.InternalLoginModule;
+import org.zanata.security.permission.CustomPermissionResolver;
 import org.zanata.security.permission.MultiTargetList;
+import org.zanata.util.Event;
 import org.zanata.util.ServiceLocator;
 
-import static org.jboss.seam.ScopeType.SESSION;
-import static org.jboss.seam.annotations.Install.APPLICATION;
+import com.google.common.collect.Lists;
 
 @Name("org.jboss.seam.security.identity")
 @Scope(SESSION)
 @Install(precedence = APPLICATION)
 @BypassInterceptors
 @Startup
-public class ZanataIdentity extends Identity {
+public class ZanataIdentity implements Identity, Serializable {
     private static final Logger log = LoggerFactory.getLogger(
             ZanataIdentity.class);
 
     public static final String JAAS_DEFAULT = "default";
+    public static final String ROLES_GROUP = "Roles";
 
     private static final long serialVersionUID = -5488977241602567930L;
 
+    protected static boolean securityEnabled = true;
+    // Seam Context variables
+    // TODO [CDI] revisit this
+    private static final String LOGIN_TRIED = "org.jboss.seam.security.loginTried";
+    private static final String SILENT_LOGIN = "org.jboss.seam.security.silentLogin";
```

Comment:
Okay, but please try to make TODO comments self-explanatory.  We have to read them too!